src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -640,7 +640,7 @@ namespace OpenMS
           pe = pes[0];
         }
 
-        f << "\t\t<search_hit hit_rank=\"1\" peptide=\""
+        f << "\t\t<search_hit hit_rank=\"" << String(h.getRank() + 1) << "\" peptide=\"" // rank in pepXML is 1-based, 0-based in OpenMS
```

Comment:
So this was never written to pepxml anyway?

Reply:
yep